library/lua/gui/widgets.lua

```diff
@@ -2433,4 +2433,131 @@ function RangeSlider:onRenderBody(dc, rect)
     end
 end
 
+GraphicButton = defclass(GraphicButton, ResizingPanel)
+GraphicButton.ATTRS {
+    tiles = DEFAULT_NIL,
+    height = 0,
+    width = 0,
+    graphics = '',
+    graphic_file_dims = { x = 0, y = 0 },
+    pos = 0,
```

Comment:
for parent-specified required values, it's better to default to `DEFAULT_NIL` and then validate in `init`